ledger/eval/eval.go

```diff
@@ -1003,7 +1003,7 @@ func (eval *BlockEvaluator) TestTransaction(txn transactions.SignedTxn) error {
 
 	// Transaction already in the ledger?
 	txid := txn.ID()
-	err = eval.state.checkDup(txn.Txn.First(), txn.Txn.Last(), txid, ledgercore.Txlease{Sender: txn.Txn.Sender, Lease: txn.Txn.Lease})
```

Comment:
idk, the wisdom of getters is that users cannot modify the underlying field

Reply:
But they were nearly unused. Just one call, I think.  Most code uses FirstValid,LastValid and nobody was using this behind an interface, so the field remains modifiable.

Reply:
I found 24 read only uses of `FirstValid` and `LastValid` in NON-test code, and only 2 that use First, Last.

If it weren't for the awkwardness of changed name, it probably wouldn't bother be so much, but it can't be `FirstValid()` because of the name clash, right?

There's an annoying example where we have `BlockHeader` with a `Round`, embedded in `Block` that offers `Round()`, so I probably wouldn't want the same name even if it was allowed.